### PR TITLE
export to RDF

### PR DESCRIPTION
**Pull request by josef-janousek**

### CODE REVIEW

File: odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          StringUtils.isNoneBlank(column.getAboutUrl(), column.getPropertyUrl())) {
+        relations.putIfAbsent(column.getAboutUrl(), new HashMap<>());
+        relations.get(column.getAboutUrl()).putIfAbsent(column.getPropertyUrl(), new HashMap<>());
+        if (column.getValueUrl() == null) {
+          relations.get(column.getAboutUrl()).get(column.getPropertyUrl()).put(String.format("{%s}", column.getName()), 
+              column.getDataType());
+        }
+        else if (isColumnLink(column.getValueUrl())) {
+          relations.get(column.getAboutUrl()).get(column.getPropertyUrl()).put(column.getValueUrl(), 
+              namesColumnsAnnotatedTable.get(column.getValueUrl().substring(1, column.getValueUrl().length()-1)).getDataType());
```
Comment (tomas-knap) [inline odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java]: What the expression "substring(1, column.getValueUrl().length()-1)" stands for? This is really suspicious

File: odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            // create the predicate
+            switch (predicateEntry.getKey()) {
+              case DCTERMS_TITLE:
+                predicate = DCTERMS.TITLE;
+                break;
+              case RDF_TYPE:
+                predicate = RDF.TYPE;
+                break;
+              default:
+                predicate = factory.createIRI(predicateEntry.getKey());
```
Comment (tomas-knap) [inline odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java]: You are creating the same predicate IRI for each row, but the IRI is the same for every row! So this part should be outside of foreach (Row r: rows)

File: odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      i++;
+    }
+    
+    HashMap<String, TableColumn> namesColumnsAnnotatedTable = new HashMap<>();
+    for (TableColumn column : annotatedTable.getTableSchema().getColumns()) {
+      namesColumnsAnnotatedTable.put(column.getName(), column);
+    }
+    
+    // fetch the relations in appropriate order
+    HashMap<String, HashMap<String, HashMap<String, String>>> relations = new HashMap<>();
```
Comment (tomas-knap) [inline odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultAnnotatedTableToRDFExportAdapter.java]: This is awful, why didn't you represent Relation as a class, which has certain URL, subject and object, right?

**Comment by tomas-knap** [comment]
When testing RDF export - running RDFExportTest on books.csv-extended and json (the original one with the wrongly outputted empty/null fields - see https://github.com/odalic/sti/pull/112#issuecomment-239398364), I got an error: 

screen shot 2016-08-17 at 10 02 26

File: odalic/pom.xml [hunk 4]
```diff
@@ -146,6 +146,13 @@
 			<artifactId>commons-csv</artifactId>
 			<version>1.4</version>
 		</dependency>
+		
+		<!-- RDF4J -->
+		<dependency>
+			<groupId>org.eclipse.rdf4j</groupId>
+			<artifactId>rdf4j-runtime</artifactId>
+			<version>2.0M4</version>
```
Comment (tomas-knap) [inline odalic/pom.xml]: Use 2.0 version, which was just released

File: odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultRDFExporter.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ */
+public class DefaultRDFExporter implements RDFExporter {
+
+  /**
+   * The default export implementation. 
+   * 
+   * @see cz.cuni.mff.xrg.odalic.outputs.rdfexport.RDFExporter#export(org.eclipse.rdf4j.model.Model, org.eclipse.rdf4j.rio.RDFFormat)
+   */
+  @Override
+  public String export(Model rdfModel, RDFFormat rdfFormat) {
```
Comment (tomas-knap) [inline odalic/src/main/java/cz/cuni/mff/xrg/odalic/outputs/rdfexport/DefaultRDFExporter.java]: There should be also alternative method, which export the Model directly to a File (not writing to a String first and then writing to a file)

**Comment by tomas-knap** [comment]
My review is pending

**Comment by tomas-knap** [comment]
@josef-janousek Please use everywhere logger, do not write to standard output. See e.g. DefaultAnnotatedTableToRDFExportAdapter in this pull request, https://github.com/odalic/sti/pull/116/files#diff-65c4d0d7cbadc2bcb9ad9d1b2be81255R34
Consultation with @brodecva or @istvan-satmari may be needed to use the agreed approach for logging.

**Comment by tomas-knap** [comment]
Review done from my site, @josef-janousek please address the 3 comments above.
I adjusted a bit certain parts of code, if you have questions, ask.

**Comment by josef-janousek** [comment]
Comments addressed.

**Pull request merged**